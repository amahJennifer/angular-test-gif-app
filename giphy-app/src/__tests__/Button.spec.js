import React from "react";
import { create } from "react-test-renderer";
import Search from "../Components/search";
import GifContainer from "../Components/gifContainer";
import { render, unmountComponentAtNode } from "react-dom";
import { act } from "react-dom/test-utils";

let container;

beforeEach(() => {
	// setup a DOM element as a render target
	container = document.createElement("ul");
	document.body.appendChild(container);
});

afterEach(() => {
	// cleanup on exiting
	unmountComponentAtNode(container);
	container.remove();
});

describe("Search Component", () => {
	test("Matches the snapshot", () => {
		const button = create(<Search />);
		expect(button.toJSON()).toMatchSnapshot();
	});

	it("renders list of Gif", async () => {
		const fakeUser = {
			name: "Joni Baez",
			age: "32",
			address: "123, Charming Avenue",
		};

		jest.spyOn(global, "fetch").mockImplementation(() =>
			Promise.resolve({
				json: () => Promise.resolve(fakeUser),
			})
		);

		// Use the asynchronous version of act to apply resolved promises
		await act(async () => {
			render(<GifContainer items={[]} />, container);
		});

		// expect(container.querySelector("Joni Baez").textContent).toBe(fakeUser.name);

		// remove the mock to ensure tests are completely isolated
		global.fetch.mockRestore();
	});
});
