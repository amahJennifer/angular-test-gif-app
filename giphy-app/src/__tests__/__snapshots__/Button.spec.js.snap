// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component Matches the snapshot 1`] = `
<form
  className="bg-dark py-5 shadow fixed"
>
  <div
    className="container px-5 py-5"
  >
    <input
      aria-label="Search"
      className="form-control mr-sm-2"
      placeholder="Search gif"
      type="text"
    />
    <div />
  </div>
  <div
    className="container text-center"
  >
    <button
      className="px-5 text-center btn btn-outline-light"
      type="submit"
    >
      Search
    </button>
  </div>
</form>
`;

exports[`Search Component  1`] = `
<form
  className="bg-dark py-5 shadow fixed"
>
  <div
    className="container px-5 py-5"
  >
    <input
      aria-label="Search"
      className="form-control mr-sm-2"
      placeholder="Search gif"
      type="text"
    />
    <div />
  </div>
  <div
    className="container text-center"
  >
    <button
      className="px-5 text-center btn btn-outline-light"
      type="submit"
    >
      Search
    </button>
  </div>
</form>
`;

exports[`Search Component Displays list of g 1`] = `
<form
  className="bg-dark py-5 shadow fixed"
>
  <div
    className="container px-5 py-5"
  >
    <input
      aria-label="Search"
      className="form-control mr-sm-2"
      placeholder="Search gif"
      type="text"
    />
    <div />
  </div>
  <div
    className="container text-center"
  >
    <button
      className="px-5 text-center btn btn-outline-light"
      type="submit"
    >
      Search
    </button>
  </div>
</form>
`;

exports[`Search Component Matches the snapshot 1`] = `
<form
  className="bg-dark py-5 shadow fixed"
>
  <div
    className="container px-5 py-5"
  >
    <input
      aria-label="Search"
      className="form-control mr-sm-2"
      placeholder="Search gif"
      type="text"
    />
    <div />
  </div>
  <div
    className="container text-center"
  >
    <button
      className="px-5 text-center btn btn-outline-light"
      type="submit"
    >
      Search
    </button>
  </div>
</form>
`;
